odl/discr/diff_ops.py

```diff
@@ -559,11 +559,13 @@ def _call(self, x, out=None):
 
         ndim = self.range.ndim
         dx = self.range.cell_sides
+        impl = self.range.impl
 
-        tmp = np.empty(out.shape, out.dtype, order=out.space.default_order)
-        with writable_array(out) as out_arr:
+        tmp = array_module(impl).empty(
+            out.shape, out.dtype, order=out.space.default_order)
```

Comment:
This of course assumes that `impl` has the same interface as numpy. Perhaps it's better to implement wrappers for the most common functions, like `asarray` (already done), `empty`, and whatever else is needed. That would be a bit more extensible.

Reply:
That would be how e.g. Keras does it, but I say we leave this for now. Changing should be rather easy (just search replace basically).